addon/metrics-adapters/google-analytics.js

```diff
@@ -55,21 +55,18 @@ export default BaseAdapter.extend({
   },
 
   trackEvent(options = {}) {
-    const compactedOptions = compact(options);
-    const sendEvent = { hitType: 'event' };
-    let gaEvent = {};
-
-    if (compactedOptions.nonInteraction) {
-      gaEvent.nonInteraction = compactedOptions.nonInteraction;
-      delete compactedOptions.nonInteraction;
-    }
-
-    for (let key in compactedOptions) {
-      const capitalizedKey = capitalize(key);
-      gaEvent[`event${capitalizedKey}`] = compactedOptions[key];
-    }
+    let compactedOptions = compact(options);
+    let eventData = { hitType: 'event' };
+
+    ['category', 'action', 'label', 'value'].forEach((key) => {
+      if (compactedOptions[key] !== undefined && compactedOptions[key] !== null) {
```

Comment:
@poteto Instead of just changing `compactedOptions[key] != undefined` to `compactedOptions[key] !== undefined` I wanted to make sure we're still checking null, and figured it's best to be explicit about it.